---
layout: post
title: "Polymorphism, genealogy, simulation, and hypothesis testing"
date: 2015-04-01
---
Modern geneticists make sense out of DNA sequences using POLYMORPHISM DATA. These data include the genotypes of many individuals sampled at one or more loci; a locus is considered polymorphic if two or more distinct types of sequences are observed, regardless of their frequencies. The goal of much modern genetic research is to find gene variations that contribute to disease. Finding these genes should allow an understanding of the disease process, so that methods for preventing and treating the disease can be developed.

<b>Why do we  care about variations in single sites (SNPs) and multiple sites (haplotypes) on DNA sequences?</b>

A <b>SNP</b> is a site in the DNA where different chromosomes differ in the base they have. For example, 30 percent of the chromosomes may have an A, and 70 percent may have a G. These two forms, A and G, are called variants or alleles of that SNP. An individual may have a genotype for that SNP that is AA, AG, or GG. When chromosomes from two random people are compared, they differ at about one in 1000 DNA sites. Thus when two random haploid genomes are compared, or all the paired chromosomes of one person are compared, there are about three million differences. When more people are considered, they will differ at additional sites. The number of DNA sites that are variable (SNPs) in humans is unknown, but there are probably between 10 and 30 million SNPs, about one every 100 to 300 bases. Of these SNPs, perhaps four million are common SNPs, with both alleles of each SNP having a frequency above 20 percent.

The majority of SNPs are not actual functional variants that contribute to the risk of getting a disease, but they can be useful as markers for finding functional variants. To find the regions with genes that contribute to a disease, the frequencies of many SNP alleles are compared in individuals with and without the disease (case and control). When a particular region has SNP alleles that are more frequent in individuals with the disease than in individuals without the disease, those SNPs and their alleles are said to be associated with the disease. These associations between a SNP and a disease indicate that there may be genes in that region that contribute to the disease.

A <b>haplotype</b> is a set of DNA variations, or polymorphisms, that tend to be inherited together. A haplotype can refer to a combination of alleles or to a set of single nucleotide polymorphisms (SNPs) found on the same chromosome. For diseases with a relatively straightforward genetic basis, the single-gene disorders, current methods are usually sufficient to find the genes involved. Most people, however, do not have single-gene disorders, but develop common diseases such as heart disease, stroke, diabetes, cancers or psychiatric disorders, which are affected by many genes and environmental factors. This is the major reason why we care about haplotypes.

Theoretically there could be many haplotypes in a chromosome region, but recent studies are typically finding only a few common haplotypes. Consider the example below, of a region where six SNPs have been studied; the DNA bases that are the same in all individuals are not shown. The three common haplotypes are shown, along with their frequencies in the population. The first SNP has alleles A and G; the second SNP has alleles C and T. The four possible haplotypes for these two SNPs are AC, AT, GC, and GT. However, only AC and GT are common; these SNPs are said to be highly associated with each other (i.e., linkage disequilibrium: AC and GT are more likely to be at the same chromosome than AT and GC).
<br/><img alt="haplotype" src="https://cloud.githubusercontent.com/assets/5496192/7121553/3f6b37b8-e1e3-11e4-9cbe-76aa3d15b2c7.jpg" />

The cost of genotyping is currently too high for whole-genome association studies that would look at millions of SNPs across the entire genome to see which SNPs are associated with disease. If a region has only a few haplotypes, then only a few SNPs need to be typed to determine which haplotype a chromosome has and whether the region is associated with a disease. In the example below, typing two SNPs is all that is needed to distinguish among the three common haplotypes. The two SNPs indicated by arrows are one pair of several possible pairs of SNPs that distinguish among the three common haplotypes.

<b>Why do we care about the geneaological information?</b>

First of all, we need to make sure the disease is genetic rather than due to some environmental factors. Among genetic causes to various haplotype and allele frequencies, there are still many possibilities. Haplotype and allele frequencies may be affected by cellular-level processes such as mutation, recombination, and gene conversion, as well as by population-level processes such as natural selection against alleles that contribute to disease. When genes are close together and associated, then selection that changes the frequency of an allele at one gene results in similar changes in the frequencies of alleles at other genes on the same haplotype. There are also demographic and social factors: population history factors such as population size, bottlenecks or expansions of population size, founder effects, isolation of a population or admixture between populations, and patterns of mate choice. 

<p>Population and evolutionary genetics study evolutionary forces or ancient events that create and maintain genetic variation. At the heart of it is the use of gene genealogy, a tree structure describes the evolutionary history (e.g., divergence) of a particular set of DNA sequences and their relatedness. The shape of the genealogy depends on genetic events such as inheritage, mutations, recombination, as well as population events such as isolation, migration, population growth, etc. geneticists have been using it to infer or predict DNA sequence variation. Due to the uncertainty elements of reproduction, the central part of genealogical data analysis is a stochastic characterization of the genealogies that relate the sequences.</p>

<p>Geneticists use two basic approaches: linkage analysis, which can be also described as identity by descent, and association analysis, which can be described identity by state. Linkage analysis looks to see if the same chromosome is inherited by the descendants of an individual that have the same trait. Association analysis looks to see if the specific gene variations are more common among people with a trait. In this example an individual can have an A or G nucleotide at a specific DNA position. One can see that the affected individuals are more likely to have a G than an A. The G nucleotide is said to be associated with being affected.</p>

<b>Why Simulation?</b>

<blockquote>Since history is played only once, we generally have distorted perceptions of social behavior, overemphasizing what actually happened, to the detriment of what could have happened. -- Duncan Watts </blockquote>

The two main reasons for wanting to simulate genetic data are, first, to gain insight into the effects that underlying demographic and mutational parameters may have on the genetic data one sees, and, secondly, to create test datasets for assessing the power of alternative genetic analysis methods. For example, we may want to generate a data set characterizing a specific population by defining population size, evolutionary time, mutation rate, etc. There is <a href="http://www.nature.com/nrg/journal/v13/n2/full/nrg3130.html">a recent review on this area</a> from a geneticist point of view. The article also points out several other review articles, which, according to the author, focus more on technical aspects.</p> 

Modern genetics studies use data collected from natural populations. This approach has two problems. Firstly, there is no replication of the ‘experiment’, only one run of evolution is available to be studied. Secondly, the starting conditions of the ‘experiment’ are unknown. However, statistical hypothesis testing expects and utilizes inherent uncertainty/randomness of observations. Imagine that we sequence a 10-kb region in the DNA chromosomes of 30 randomly chosen individuals and, surprisingly, find no polymorphisms. We might interpret this observation as evidence for selective constraint in this region. Alternatively, it might be that the individuals chosen for the comparison are unusually closely related. The interpretation depends on the genealogy of the sampled sequences, which is a tree structure describing the unique, often unknown history of mutation, recombination and COALESCENCE of lineages in the ancestry of the sample. 

One solution is to model and simulate the past using a suitable stochastic model. To decide if the above data are unusual, we might make assumptions about the process that gave rise to those data, model the evolutionary process, and then run the simulated process many times to generate random repetitions. If the fraction of the random genealogical and mutational histories that could have given rise to the observed data is small, we can conclude that the assumptions cannot explain the observed pattern. Thus, the key is to find suitable models that allow us to construct random genealogies as a result of certain evolutionary forces.

<b>Why the coalescent (backward) approach?</b>

All simulation algorithms face three primary challenges: (1) speed — typically one wants to do lots of simulations, so they need to be fast; (2) scalability — with the advent of genome-wide genotyping and large-scale sequencing, there is a need for simulation programs to match; and (3) flexibility — can the program cope with different demographic histories, population structure, recombination, selection, mutation models and disease models? Three approaches (termed ‘backwards’, ‘forwards’ and ‘sideways’) have been developed to deal with these challenges.

The ‘backwards’ (or coalescent) approach (Kingman 1982, Hudson 1983 & 1990, Donnelly and Tavare 1995) is an efficient way to sample sequences from a theoretical population that follows the Wright-Fisher neutral model (Ewens 1979). It starts with the sample of individuals that will form your simulated data-set, then work backwards in time to construct the ancestral tree or graph of genealogical relationships that connects them all. Neutral mutations can subsequently be placed on this structure to create the simulated data-set. ‘Forwards’ simulations start with the entire population of individuals — typically, many thousands — and then follow how all the genetic data in question are passed on from one generation to the next. The coalescent approach is more efficient than the ‘forwards’ approach by restricting attention just to the genealogical structure relevant to the sample in question, that is, ignoring sequences that become extinct in the middle or whose descendent are not in the sample (see the figure below).
<br/><img alt="coalescentEfficient" src="https://cloud.githubusercontent.com/assets/5496192/7167720/b2475c78-e385-11e4-9765-4b5c7ddaa6a0.PNG">

Also, the coalescent approach usually employs a continuous-time approximation to effectively skip over the intermediate generations between important tree generating events, which makes it even more efficient. Moreover, the coalescent approach avoids the difficulty and ambiguity in choosing suitable initial conditions. When using the 'forward' approach, one usually needs to simulate over many thousands of generations in order to arrive at an equilibrium in which the genetic characteristics of the population are independent of the original starting conditions.

<b>What specific hypotheses are we testing via the coalescent approach and how?</b>
One of the most important questions we face is: Is the observed DNA sequence data consistent with neutral evolution
in a homogeneously mixing population of constant size? As you can read from <a href="http://lybird300.github.io/2015/04/02/coalescent.html">my next post</a>, this null hypothesis can be violated in many aspects. Alternative hypotheses could comprise either of the following or their combinations:
<ul>
<li>Population bottleneck. If, as we work backwards in time, there is a sudden decrease in the population size, then the coalescence rate will become large. Situations that can cause this are the founding of a new population by a small number of migrants or, what is essentially the same, improving a crop species by choosing a few individuals with desirable properties.</li>
<li>Selective sweep. This term refers to the appearance of a favorable mutation that rises in frequency until it takes over the population. In the absence of recombination, this is a severe population bottleneck because the entire population will trace its ancestry to the individual with the favorable mutation.</li>
<li>Population growth. In the human population, which has experienced a period of exponential growth, then the coalescence rate will be initially small, and the genealogical tree will have tips that are longer than usual. Population subdivision. Imagine two isolated populations that exchange migrants infrequently. If we sample 10 individuals from each population, then the two subsamples will coalesce as usual and then eventually the their two ancestral lineages will coalesce.</li>
<li>Population subdivision. Imagine two isolated populations that exchange migrants infrequently. If we sample 10 individuals from each population, then the two subsamples will coalesce as usual and then eventually the their two ancestral lineages will coalesce.</li>
<li>Balancing selection. This term refers to a situation in which the fitness of heterozygotes Aa is larger than that of homozygotes AA and aa. In this case, the population will settle into a equilibrium in which the two alleles A and a are each present with fixed frequencies. This is essentially a two-population model, with migration between chromosomes with A and chromosomes with a being caused by mutation or recombination.</li>
</ul>
The above alternatives lead to changes in the genealogical tree or graph (if recombination is considered).

As for future research interests, the simulation of <b>copy number variation</b> and/or microsatellite data at larger genomic scales, and of more complex disease models allowing covariates and linked loci, remain areas that deserve further exploration. (Note:  the performance of any methods trying to tackle the challenges of complex disease mapping should be evaluated by large scale simulation studies)



<h2>References</h2>
<ul>
<li><a href="http://www.genome.gov/10001665">Developing a Haplotype Map of the Human Genome for Finding Genes Related to Health and Disease</a>. Washington, D.C. July 18-19, 2001</li>
<li>Liu, Youfang, Georgios Athanasiadis, and Michael E. Weale. "A survey of genetic simulation software for population and epidemiological studies." Hum Genomics 3.1 (2008): 79-86.</li>
<li>Hoban, Sean, Giorgio Bertorelle, and Oscar E. Gaggiotti. "Computer simulations: tools for population and evolutionary genetics." Nature Reviews Genetics 13.2 (2012): 110-122.</li>
<li>Rosenberg, Noah A., and Magnus Nordborg. "Genealogical trees, coalescent theory and the analysis of genetic polymorphisms." Nature Reviews Genetics 3.5 (2002): 380-390.</li>
</ul>
