---
layout: post
title: "Linkage Disequlibrium (LD) mapping and coalescent"
date: 2015-04-19
---
First, 
<h2>IBD (Identical By Descent) vs IBS (Identical By State)</h2>
The term IBS, Identical by State, refers to the situation that two DNA segments are identical at all possible allele positions, no matter how that matching occurred. The two segments can be IBD (if the matching alleles are inherited from a common ancestor in a genealogical timeframe) or not (if the segment is ubiquitous within a specific population). To understand the latter situation, think about right/left-handedness. Ten percent of the population are left-handed. If you are right-handed and meet another person that is right-handed, does that mean you are relatives? Of course not. You might further argue that IBS segments essentially IBD, because every allele that modern people (Homo sapiens sapiens) got from Lucy’s people (Australopithecus afarensis). True, but when we talk about IBD, we often restrict ourselves to a genealogical time frame (about 500 to possibly slightly more than 1000 years, or the last 20 generations).

There are several ways certain sequences become very common in a population. That sequence could be very old and because it provided a very slight advantage to those that had it, it remained intact for thousands of years. But because the advantage was small enough, those that didn’t have the exact same sequence could also thrive and reproduce but at a slightly lower rate. Another way is when a population goes through an evolutionary bottleneck. This is where the population gets very small for whatever reason — disease, famine, or isolation such as when small group migrates to a new land — and then expands to fill its niche again. When this happens, the assortment of alleles that are available is sharply reduced. In these cases, many people can have similar alleles that are IBS but their common ancestor would have to be traced all the way back to the bottleneck. When the bottle neck is far enough back that no (or few) genealogical records persist, those segments are now considered IBS.

If your father and your mother both have the same sequence one of their chromosome pairs and both you and your sister inherit one copy of that sequence from different parents, then that segment is IBS between you and your sister and not IBD. But <b>there is no way to determine that happened</b>. The more similar or inbred the population that your parents come from were, the more likely this is to happen. Also note that current DNA typing techniques cannot say which one of the two chromosomes each allele came from. For example, if you are <b>A</b>A <b>G</b>G <b>A</b>A <b>C</b>C you will match me when I am <b>A</b>G <b>A</b>G <b>A</b>G <b>C</b>T, but there is no guarantee that A G A C came from the same chromosome (e.g., it could be A--G on one chromosome and A--C on the other). To be more confident in our conclusion, a much longer match is often needed. 





rare events, need fine-scale mapping (IBD help find finer region) and powerful statistical test

The technique known as association mapping differs from traditional pedigree studies in that marker-disease associations
are sought in populations of unrelated individuals. To explain the idea behind this approach, imagine that a disease causing mutation has just occurred in a population. The chromosome on which this mutation occurred contains specific alleles in neighboring polymorphic loci. At first, the mutation will only be observed in conjuction with these alleles, so the association with these alleles will be high. Through time these associations will dissipate because of recombination, but the closest loci will experience the fewest recombinations and hence retain the highest levels of LD. Thus, by looking for significant correlations between disease state and alleles, we can hope to identify the region in which the disease causing genetic mutation lies.

Linkage disequilibrium (LD) refers to the nonrandom association of alleles in haplotypes. Such associations underlie all forms of genetic mapping (a process of establishing the locations of genes on the chromosomes), including the fine scale mapping of human disease loci.

Genetic Linkage occurs when particular genetic loci or alleles for genes are inherited jointly. Genetic loci on the same chromosome are physically connected and tend to segregate together during meiosis, and are thus genetically linked.

Linkage Mapping has been successfully been applied for the identification of a large number of rare, single gene human diseases. However, the resolution of the method is determined by the number of meiosis oberved in pidigree, which is not very big number.
Recombination rate will also be very low in the nearby region.
(Limited meiosis)
The region is likely to cover several Mb of DNA sequence and
to contain several hundred genes.


whereas linkage analysis is based upon associations in well-characterized pedigrees, LD refers to associations
within populations of ‘unrelated’ individuals. There
is nonetheless a close relationship between the two
approaches, because the ‘unrelated’individuals in
a population are unrelated only in a relative and
approximate sense. In general, chromosomes sampled
from ‘unrelated’individuals in a population will be
much more distantly related than those sampled from
members of traditional pedigrees. 

This is precisely
what makes LD mapping suitable for fine-scale
mapping: there will have been more opportunities
for recombination to take place. Whereas pedigree
studies work with recombination events that
exchange megabase chunks of chromosomes, LD
studies deal with segments measured in kilobases.  When dealing with unrelated individuals,
however, the pedigree is almost completely unknown.

For human populations
might be 4,000 and 40,000 generations. It must contains
more meiosis.
The more meiosis, and the more the recombination happens,
and the shorter the relative region is.
A number of genetic markers are typed at known position.
It becomes possible, in principle at least, to map a trait of
interest much more accurately.


Genetic polymorphism is the result of mutations along the branches of genealogical trees. 

All existing copies of a unique
mutation can be traced back to the most-recent common
ancestor (MRCA) of that mutation. Each haplotype that
contains the mutation must also have inherited from the
MRCA a small piece of chromosome surrounding the
mutation. Different haplotypes might carry different
pieces, depending on the history of recombination. Alleles at polymorphic marker loci within the region
covered by these pieces of chromosome will be in LD with
the mutation.

Recombination is the major process that breaks down the associations between SNPs. It is unclear whether haplotype block boundaries are due to recombination hotspots, or are simply the result of recombination events that happened to occur there. If the blocks are due to hotspots, then perhaps they will be common across populations. If the blocks are due to regular recombination events, then populations may or may not share them, depending on how long ago the recombination events occurred. When large chromosomal regions are examined, the regions with high association have less recombination and less genetic variation. 

SNPs are the preferred marker type of LD mapping because of their abundancy in the genome, however, large differences in the
density of SNPs exist over the genome with regard to the genealogical process, mutation rate, selective regimes, and the  recombination and gene conversion rates per base pair.Considering whole haplotypes consisting of several adjacent
markers rather than single markers one at a time

None of the markers are individually strongly associated with
phenotype.

