---
layout: post
title: "Linkage Disequlibrium (LD) mapping and coalescent"
date: 2015-04-19
---

Linkage disequilibrium (LD) refers to the nonrandom association of alleles in haplotypes. Such associations underlie all forms of genetic mapping (a process of establishing the locations of genes on the chromosomes), including the fine scale mapping of human disease loci.

Genetic Linkage occurs when particular genetic loci or alleles for genes are inherited jointly. Genetic loci on the same chromosome are physically connected and tend to segregate together during meiosis, and are thus genetically linked.

Linkage Mapping has been successfully been applied for the identification of a large number of rare, single gene human diseases. However, the resolution of the method is determined by the number of meiosis oberved in pidigree, which is not very big number.
Recombination rate will also be very low in the nearby region.
(Limited meiosis)
The region is likely to cover several Mb of DNA sequence and
to contain several hundred genes.


whereas linkage analysis is based upon associations in well-characterized pedigrees, LD refers to associations
within populations of ‘unrelated’ individuals. There
is nonetheless a close relationship between the two
approaches, because the ‘unrelated’individuals in
a population are unrelated only in a relative and
approximate sense. In general, chromosomes sampled
from ‘unrelated’individuals in a population will be
much more distantly related than those sampled from
members of traditional pedigrees. 

This is precisely
what makes LD mapping suitable for fine-scale
mapping: there will have been more opportunities
for recombination to take place. Whereas pedigree
studies work with recombination events that
exchange megabase chunks of chromosomes, LD
studies deal with segments measured in kilobases.  When dealing with unrelated individuals,
however, the pedigree is almost completely unknown.

For human populations
might be 4,000 and 40,000 generations. It must contains
more meiosis.
The more meiosis, and the more the recombination happens,
and the shorter the relative region is.
A number of genetic markers are typed at known position.
It becomes possible, in principle at least, to map a trait of
interest much more accurately.


Genetic polymorphism is the result of mutations along the branches of genealogical trees. 

All existing copies of a unique
mutation can be traced back to the most-recent common
ancestor (MRCA) of that mutation. Each haplotype that
contains the mutation must also have inherited from the
MRCA a small piece of chromosome surrounding the
mutation. Different haplotypes might carry different
pieces, depending on the history of recombination. Alleles at polymorphic marker loci within the region
covered by these pieces of chromosome will be in LD with
the mutation.

Recombination is the major process that breaks down the associations between SNPs. It is unclear whether haplotype block boundaries are due to recombination hotspots, or are simply the result of recombination events that happened to occur there. If the blocks are due to hotspots, then perhaps they will be common across populations. If the blocks are due to regular recombination events, then populations may or may not share them, depending on how long ago the recombination events occurred. When large chromosomal regions are examined, the regions with high association have less recombination and less genetic variation. 

SNPs are the preferred marker type of LD mapping because of their abundancy in the genome, however, large differences in the
density of SNPs exist over the genome with regard to the genealogical process, mutation rate, selective regimes, and the  recombination and gene conversion rates per base pair.Considering whole haplotypes consisting of several adjacent
markers rather than single markers one at a time

None of the markers are individually strongly associated with
phenotype.

