---
layout: post
title: "Genome maps and genetic mapping"
date: 2015-04-21
---
Like a physical map, a genome map tells people where they are and helps them get where they want to go. It contains a set of landmarks that helps scientists navigate around the genome, such as short DNA sequences, regulatory sites that turn genes on and off, and genes themselves. A genome map today is nothing close to a highly precise road map though. It is more like a map of North America made when Europeans were just beginning to explore the continent. Some parts of the genome have been mapped in great detail, while others remain relatively uncharted territory. It may turn out that a few landmarks on current genome maps appear in the wrong place or at the wrong distance from other landmarks. But over time, as scientists continue to explore the genome frontier, maps will become more accurate and more detailed.

Most everyday maps have length and width, latitude and longitude, like the world around us. But a genome map only has one dimension. It is linear, like the DNA molecules that make up the genome itself. A genome map looks like a straight line with landmarks noted at irregular intervals along it, much like the towns along the map of a highway. The landmarks are usually inscrutable combinations of letters and numbers that stand for genes or other featuresâ€”for example, D14S72, GATA-P7042, and so on. Thus, a genome map is close to a genome sequence in portraying a genome, but they are different. Consider the following sequence:
<br/><img alt="sequence" src="https://cloud.githubusercontent.com/assets/5496192/7442123/975c74ee-f0d1-11e4-9da1-b0ee862a3e29.gif" />
A map of that sequence might look like this:
<br/><img alt="map" src="https://cloud.githubusercontent.com/assets/5496192/7442122/9531b8f0-f0d1-11e4-97b4-8f8b1ef08765.gif" />
A genome map simply identifies a series of landmarks in the genome; for example, GCC and CCCC on the map above. In contrast, the corresponding sequence spells out the order of every DNA base in the genome. In this sense, the sequence is the most detailed possible map. It also implies that we can determine the location of a gene (i.e., to "map" the gene) without <b>sequencing</b> it, but by use of certain <b>mapping</b> techniques (discussed below). Because mapping involves less information to collect and organize than sequencing does, creating a reasonably comprehensive genome map is generally quicker and cheaper than sequencing the entire genome, particularly for humans and other species with large genomes. That said, the ideal genome map is one that depends on physical distance in the DNA sequence measured in base pairs (bp) or kilobases (kb). This type of genome maps is thus referred to as the physical map. 

Studying the human genome is therefore a two-pronged effort, aiming at both a comprehensive genome map and a complete genome sequence. At first glance this strategy may seem redundant, since a sequence is simply the most detailed map possible. Why not just sequence the genome? Why keep mapping the human genome if it's already been sequenced? One reason is that a map can actually help you sequence the genome. If you're sequencing a genome with the clone-by-clone method, you need a map to determine where each clone belongs in the genome. The more detailed and accurate your map, the easier it is to snap those pieces of genomic jigsaw puzzle into place. With whole-genome shotgun sequencing, a map is no longer central to the strategy, but one can still be used to help match the big pieces of assembled sequence to their proper place in the genome. 

Another reason is that you need a map to understand the genome sequence. As a map may tell you nothing about the sequence of the genome, a sequence may tell you nothing about the map. A sequence is just a long, long string of DNA bases or "letters." For the most part, scientists can't look at a sequence and see immediately which parts are genes or other interesting features, and which parts are "junk." But the landmarks on a genome map provide clues about where the important parts of the genome sequence can be found.

Linear arrangement implies not merely order of loci but the additivity of their distances. The first type of genomic maps, also known as the linkage map (take a look at the interesting and inspiring story behind the <a href="http://www.discoveryandinnovation.com/BIOL202/notes/lecture7.html">construction of the first genetic map</a>) use a unit of recombination frequency to indicate distance along a chromosome. The unit is called centimorgan (abbreviated cM) or map unit (m.u.). One centimorgan is equal to a 1% chance that a marker at one genetic locus on a chromosome will be separated from a marker at a second locus due to crossing over in a single generation. The number of base-pairs 1 centimorgan corresponds to varies widely across the genome (different regions of a chromosome have different propensities towards crossover). In human genome, 1 centimorgan on average represents a distance of about 7.5x10E5 base pairs.A second type of genetic maps is known as the linkage disequilibrium map. It determines the distance between pairs of loci not from recombination frequency but from allelic association, also called linkage disequilibrium (LD), which is measured in linkage disequilibrium units (LDU). One LDU corresponds to the length of a chromatid in which on the average one crossover event has taken place in t generations, and so the resolution is t times as great as for the linkage map. This greatly increases the power to localize a gene within a candidate region at the expense of typing a larger number of markers and using methods that at this early stage are less transparent to many geneticists. With the essential properties of linearity and additivity, LD mapping is a recent invention still in rapid evolution.
 










<h2>References</h2>
<ul>
<li><a href="http://www.discoveryandinnovation.com/BIOL202/index.html">BIOL 202 Genetics taught by Dr. Paul SzauterUniversity of New Mexico</a>. Lecture note 7. </li>
<li>Morton, Newton E. "History of genetic mapping." Encyclopedia of Genetics, Genomics, Proteomics and Bioinformatics (2004).</li>
<li><a href="http://www.genomenewsnetwork.org/resources/whats_a_genome/Chp3_1.shtml">GENOME MAP</a>. Genome News Network</li>
</ul>
