---
layout: post
title: "Detection of identity-by-descent (IBD) segments in population data"
date: 2015-08-31
---
An identity-by-descent (IBD) segment is a continuous segment that delinearates IBD haplotypes, i.e., haplotypes shared by inheritance from a recent common ancestor. The neutral coalescent theory in population genetics implies that all individuals have common ancestry in the distant past. Any given pair of individuals is related through many common ancestors. IBD segment detection intends to find out the most recent, single common ancestor from whom an IBD segment is inherited. Thus, when detecting an IBD segment in population data, the IBD segment must have sufficient length to provide confidence that the segment is not a fusion of multiple short IBD segments from different distant/ancient common ancestors (although some error is often allowed in identifying the segment endpoints). This length constraint is defined as the resolution of IBD segment detection for a data set and it may vary with specific data or detection algorithms.

Closely related individuals have a high proportion of IBD, and their IBD is arranged in long continuous segments. For example, halfsiblings have, on average, IBD sharing across one half of the genome, and the average IBD segment length is 50 centiMorgans (cM) or approximately 50 megabases (Mb). As the relationship becomes more distant, the average proportion of the genome shared identical by descent decreases exponentially, but the average length of an IBD segment, when IBD sharing do occur, remains relatively long. For a pair of individuals separated by m meioses, the average proportion of the genome shared identical by descent (the sum of lengths of IBD segments divided by the length of the genome) owing to that shared ancestry is 1/(2^(m−1)). In contrast, the lengths of IBD segments owing to the shared ancestry are approximately exponentially
distributed with a mean of 100/m cM. In other words, individuals who have shared ancestry through a certain common ancestor 25 generations ago (with 50meioses of separation) almost always share none of their genome identical by descent through that ancestor, but if they do have an IBD segment through that ancestor it will have a mean length of 2 cM (∼2 Mb). Further, although any given pair of unrelated individuals has very little IBD sharing, the total amount of IBD sharing in a big data set with many unrelated individuals (say N, then there will be N(N-1)/2 pairs of individuals), and the total amount per individual, can be large.

The key idea behind IBD segment detection is haplotype frequency. One can infer the presence of IBD segments from haplotypes that have low frequency and therefore are rarely observed in multiple independently sampled individuals. This criterion can be applied in several ways. The first is through the length of sharing, which is a proxy for frequency. If two densely genotyped haplotypes are identical at all or most (allowing for some genotyping error) assayed alleles over a very large segment of a chromosome, then the two haplotypes are likely to be identical by descent across the whole segment. The second is a direct use of haplotype frequency: Shared haplotypes with estimated frequency below some threshold are determined to be identical by descent. The third makes use of a population genetics model to infer probability of IBD. Given the frequency of the shared haplotype and a probability model for the IBD process along the chromosome, one can estimate the probability that the individuals are identical by descent at any position on the segment. Unfortunately, empirical data are usually not at the haplotype level but unphased genotypes and we have to estimate haplotypes from genotype data (called haplotyping or phasing). Although most methods allow for errors in haplotype estimation, such errors significantly reduce the power to detect an IBD segment. Therefore, stringent data quality control should be applied beforehand to remove lower-quality variants.

<h2>Specific Methods</h2>
Perhaps the simplest approach to IBD segment detection is to look at the genotypes of a pair of (or more) individuals and find long shared segments that are consistent with IBD. Genotypes are consistent with IBD if they have at least one allele in common. For biallelic SNP markers, a pair of genotypes is consistent with IBD unless they are discordant homozygotes. For example, the genotype pair AA/AT is consistent with IBD, whereas the genotype pair AA/TT is not. That said, we cannot As mentioned, a large number of IBD-consistent SNPs are needed to be confident that a continuous IBD segment is present. The length threshold used depends on the desired tradeoff between false-negative and false-positive results. This problem is further exacerbated by linkage disequilibrium (LD). Discordant homozygotes may
cluster along the genome because of LD correlation,
as do genotypes consistent with IBD.
For example, when two markers are in perfect
LD (r2 = 1), concordant homozygotes at one
position are always accompanied by concordant
homozygotes at the other position. This correlation
increases the length of IBD-consistent
SNPs required to infer recent IBD.

<h2>Application</h2>
