---
layout: post
title: "IBD, IBS and related methods"
date: 2015-04-29
---
<h2>IBD vs. IBS</h2>
The term IBS (Identical by State) refers to the situation that two DNA segments are identical at all possible allele positions, no matter how that matching occurred. The two segments can be IBD (identical by descent) if the matching alleles are directly inherited from a common ancestor in a genealogical timeframe, or not if the segment is ubiquitous within a specific population. To understand the latter situation, think about right/left-handedness. Ten percent of the population are left-handed. If you are right-handed and meet another person that is right-handed, does that mean you are relatives? Of course not. You might further argue that IBS segments essentially IBD, because every allele that modern people (Homo sapiens sapiens) got from Lucy’s people (Australopithecus afarensis). True, studies have shown that “unrelated” subjects commonly have long segments of shared haplotypes and most of us are IBD for 1-2% of our genome. It is just that when we talk about IBD, we often restrict ourselves to a genealogical time frame (about 500 to possibly slightly more than 1000 years, or the last 20 generations).

There are several ways certain sequences become very common in a population. That sequence could be very old and because it provided a very slight advantage to those that had it, it remained intact for thousands of years. But because the advantage was small enough, those that didn’t have the exact same sequence could also thrive and reproduce but at a slightly lower rate. Another way is when a population goes through an evolutionary bottleneck. That is, the population rapidly shrinks for whatever reason — disease, famine, or isolation such as when small group migrates to a new land — and then expands to fill its niche again. When this happens, the assortment of alleles that are available is sharply reduced. In these cases, many people can have similar alleles that are IBS but their common ancestor would have to be traced all the way back to the bottleneck. When the bottle neck is far enough back that no (or few) genealogical records persist, those segments are now considered IBS.

Here is a tricky example. If your father and your mother have the same DNA sequence in their chromosomes and you and your sister get one copy of that sequence from different parents, then that segment between you and your sister is IBS, not IBD. But <b>there is no way to determine which one is the case</b>. The more similar or inbred the population that your parents come from, the more likely this ambiguity exists.

<h2>Linkage vs. association analysis</h2>
To identify gene variations that cause human diseases, geneticists use two basic approaches that use logic and information based on IBD and IBS respectively. Linkage analysis focuses on well-characterized pedigrees to see if the same chromosome is inherited by the descendants of an individual that have the same trait. Association analysis looks to see whether specific gene variations are more common among people who are unrelated but share a certain trait. There is nonetheless a close relationship between the two approaches, because the "unrelated" individuals in a population are unrelated only in a relative and approximate sense (as discussed in the previous section). Basically, chromosomes sampled from "unrelated" individuals in a population will be much more distantly related than those sampled from members of traditional pedigrees. As a consequence, association analysis allows for finer localization/mapping of human disease loci (i.e., genes) than linkage analysis.

To explain the idea behind, imagine that a disease causing mutation has just occurred in a population. The chromosome on which this mutation occurred contains specific alleles in neighboring polymorphic loci. At first, the mutation will only be observed in conjuction with these alleles, so the association with these alleles will be high. Through time these associations will dissipate because of recombination, but the closest loci will experience the fewest recombinations and hence retain the highest levels of nonrandom associations. Thus, by looking for significant correlations between disease state (e,g., affected or not) and genetic markers (causal variations or their close loci), we can hope to identify the region in which the disease causing genetic mutation lies. The resolution of linkage analysis is determined by the number of meiosis oberved in pedigree, which is not very big (often the pedigree information we can obtain only trace back a few generations). Limited meiosis also means the recombination rate in nearby regions will also be very low. Thus, linkage analysis is likely to discover chromosomes regions that cover several Mb of DNA sequence and contain several hundred genes. Given an unrelated population, there will have been more meiosis. The more meiosis, the more opportunities for recombination to take place, and the shorter the relative region is. Thus, association studies can, in principle at least, map a trait of interest more to a smaller and more accurate segment (measured in Kbs).

The following figure (borrowed from Kirk's Yale talk slide deck) summarizes what we have discussed so far. Linkage analysis using multiple families has been a powerful tool for identifying rare and highly penetrant genetic variants. However, most linkage studies require a large number of families with affected individuals to map the disease causing variant, and even so, the causative variant may only bemapped to a larger genomic region. In recent years genome-wide association studies (GWAS) using unrelated individuals and high throughput single nucleotide polymorphism (SNP)-chips has been successful in identifying common variants with a relative low penetrance. However, the variants identified by GWAS to date only explain 18–24% of the heritability. Many believe that much of the missing heritability is probably explained by common variants of smaller effect sizes and some heritability may be explained by rare variants of larger effect size. 
<br/><img alt="IBDvsIBS" src="https://cloud.githubusercontent.com/assets/5496192/7396806/1c8a22a4-ee72-11e4-8873-97a6e407f2bd.png" width="80%" height="%80"/><br/>

<h2>IBD Mapping</h2>
Standard analyses of GWAS data (SNP association mapping) are not designed to detect the effects of rare variants ( low frequency, i.e., MAF ≤ 5%), and other approaches have been developed for this purpose. One approach is to re-analyze GWAS data using identity-by-descent (IBD) mapping. IBD mapping is performed on unrelated individuals at the population level to detect long haplotypes (generally one to five Mbs) shared by the cases (individuals with the same trait of interest) that are also IBD (inherited from distant common ancestors 10–100 generations ago). HapMap Phase 3 identified that lower frequency variants should, on average, be younger than more common variants and thus display a greater extent of haplotype sharing. If two individuals have a common ancestor more than ten generations ago, they tend to share very short tracts of genetic material, then a long shared haplotype will be very rare and also very likely to be IBD. If case pairs can be detected with long shared haplotypes inherited from distant common ancestors, then rare variants influencing disease risk can be localized. Thus, IBD mapping helps reduce the massive multiple testing problem by prioritizing regions. Admittedly, the optimal method to detect rare disease-causing variants is whole genome sequencing of thousands of samples, However, even when this becomes affordable, there will still remain a role for IBD analysis to prioritize regions for follow-up analysis and minimize the massive multiple testing burden.

The idea of using IBD haplotype sharing to detect signals of disease-causing variants in population samples is not new, but the greatly increased density of SNP markers now makes it possible to detect much smaller segments of IBD. Now we can determine pairwise IBD sharing in a large sample over the whole genome to a resolution of approximately 2 cM (length of IBD segments). Two types of statistics have been proposed for IBD mapping. The first type use IBD detected between pairs of individuals (Purcell et al. 2007): the rate of IBD in case/case pairs is compared to the rate of IBD in either control/control pairs or non-case/case pairs (control/control and control/case pairs). The second type of statistics cluster haplotypes into IBD classes at a locus: all haplotypes within a class are IBD with each other at the locus; clusters are tested for association with case-control status. An individual person is usually a member of two IBD classes at a locus because he/she has a pair of chromosomes, although the two classes will be the same if the individual is homozygous by descent. 

Several methods of IBD mapping have been published: these include PLINK, GERMLINE, BEAGLE IBD and BEAGLE fastIBD. The models employed by PLINK and GERMLINE assume SNPs are in linkage equilibrium, so "pruning" of SNPs is required to avoid false positives due to under-estimates of population haplotype frequencies. However pruning of SNPs in incomplete linkage disequilibrium (LD) discards potentially useful information and reduces power. BEAGLE IBD and fastIBD implement a variable length Hidden Markov Model to account for LD and model haplotype frequencies more accurately. BEAGLE fastIBD runs considerably faster than BEAGLE IBD (of the order of 1000 times faster with large GWAS datasets). This is mainly because 1) it does not formally model IBD status (‘IBD’/’not IBD’) between pairs of individuals using a Hidden Markov Model as in BEAGLE IBD; 2) it stores haplotype frequencies in a data dictionary (as in GERMLINE) which means computational time scales with sample size n like nlog n instead of n^2. 

 

IBD analysis limitations: it is only suited to discover rare variants if all variants act in the same direction in one gene. For example, the identified rare variants in BRCA1 and BRCA2 gene all increase risk of breast cancer, and the four rare variants identified in IFIH1 gene all protect against type I diabetes. If some rare variants increase risk while others in the same gene decrease risk then the signal in the region will be attenuated. In addition, IBD analysis is sensitive to genotyping error, resulting in reducing signal strength. The linkage signal detected depends on a lot of markers or long haplotypes, containing up to hundreds of SNPs, a single error occurring in reading a single marker significantly reduces the signal. 



Mathematically IBD means that the probability of observing an allele in one individual is not independent from observing an allele in another individual. It is this property that is exploited in linkage analysis. Pairwise relatedness is measured in probabilities of alleles from the two individuals being IBD.



Current DNA typing techniques cannot tell which one of the two parental chromosomes (maternal or paternal) an allele inherits from, although this distinction is often important for understanding disease causality. We carry two nucleotides in each inherited position, one from Mom and one from Dad. DNA typing results for these positions is a mixture of Mom’s and Dad’s contributions in no order. For example, if you are <b>A</b>A <b>G</b>G <b>A</b>A <b>C</b>C you will match me when I am <b>A</b>G <b>A</b>G <b>A</b>G <b>C</b>T, but there is no guarantee that A G A C came from the same chromosome (e.g., it could be A--G on one chromosome and A--C on the other). This is known as the phase uncertainty.

<h2>References</h2>




