---
layout: post
title: "Genome maps and genetic mapping"
date: 2015-04-21
---
<h2>Overview</h2>
Like a physical map, a genome map tells people where they are and helps them get where they want to go. It contains a set of landmarks that helps scientists navigate around the genome, such as short DNA sequences, regulatory sites that turn genes on and off, and genes themselves. A genome map today is nothing close to a highly precise road map though. It is more like a map of North America made when Europeans were just beginning to explore the continent. Some parts of the genome have been mapped in great detail, while others remain relatively uncharted territory. It may turn out that a few landmarks on current genome maps appear in the wrong place or at the wrong distance from other landmarks. But over time, as scientists continue to explore the genome frontier, maps will become more accurate and more detailed.

Most everyday maps have length and width, latitude and longitude, like the world around us. But a genome map only has one dimension. It is linear, like the DNA molecules that make up the genome itself. A genome map looks like a straight line with landmarks noted at irregular intervals along it, much like the towns along the map of a highway. The landmarks are usually inscrutable combinations of letters and numbers that stand for genes or other featuresâ€”for example, D14S72, GATA-P7042, and so on. Thus, a genome map is close to a genome sequence in portraying a genome, but they are different. Consider the following sequence:
<br/><img alt="sequence" src="https://cloud.githubusercontent.com/assets/5496192/7442123/975c74ee-f0d1-11e4-9da1-b0ee862a3e29.gif" /><br/>
A map of that sequence might look like this:
<br/><img alt="map" src="https://cloud.githubusercontent.com/assets/5496192/7442122/9531b8f0-f0d1-11e4-97b4-8f8b1ef08765.gif" /><br/>
A genome map simply identifies a series of landmarks in the genome; for example, GCC and CCCC on the map above. In contrast, the corresponding sequence spells out the order of every DNA base in the genome. In this sense, the sequence is the most detailed possible map. It also implies that we can determine the location of a gene (i.e., to "map" the gene) without <b>sequencing</b> it, but by use of certain <b>mapping</b> techniques (discussed below). Because mapping involves less information to collect and organize than sequencing does, creating a reasonably comprehensive genome map is generally quicker and cheaper than sequencing the entire genome, particularly for humans and other species with large genomes. That said, the ideal genome map is one that depends on physical distance in the DNA sequence measured in base pairs (bp) or kilobases (kb). This type of genome maps is thus referred to as the physical map. 

Studying the human genome is therefore a two-pronged effort, aiming at both a comprehensive genome map and a complete genome sequence. At first glance this strategy may seem redundant, since a sequence is simply the most detailed map possible. Why not just sequence the genome? Why keep mapping the human genome if it's already been sequenced? One reason is that a map can actually help you sequence the genome. If you're sequencing a genome with the clone-by-clone method, you need a map to determine where each clone belongs in the genome. The more detailed and accurate your map, the easier it is to snap those pieces of genomic jigsaw puzzle into place. With whole-genome shotgun sequencing, a map is no longer central to the strategy, but one can still be used to help match the big pieces of assembled sequence to their proper place in the genome. 

Another reason is that you need a map to understand the genome sequence. As a map may tell you nothing about the sequence of the genome, a sequence may tell you nothing about the map. A sequence is just a long, long string of DNA bases or "letters." For the most part, scientists can't look at a sequence and see immediately which parts are genes or other interesting features, and which parts are "junk." But the landmarks on a genome map provide clues about where the important parts of the genome sequence can be found.

Linear arrangement implies not merely order of loci but the additivity of their distances. The first type of genomic maps, also known as the linkage map (take a look at the interesting and inspiring story behind the <a href="http://www.discoveryandinnovation.com/BIOL202/notes/lecture7.html">construction of the first genetic map</a>) use a unit of recombination frequency to indicate distance along a chromosome. The unit is called centimorgan (abbreviated cM) or map unit (m.u.). One centimorgan is equal to a 1% chance that a marker at one genetic locus on a chromosome will be separated from a marker at a second locus due to crossing over in a single generation. The number of base-pairs 1 centimorgan corresponds to varies widely across the genome (different regions of a chromosome have different propensities towards crossover). In human genome, 1 centimorgan on average represents a distance of about 7.5x10E5 base pairs.

A second type of genetic maps is known as the linkage disequilibrium map. It determines the distance between pairs of loci not from recombination frequency but from allelic association, also called linkage disequilibrium and measured in linkage disequilibrium units (LDU). One LDU corresponds to the length of a chromatid in which on the average one crossover event has taken place in t generations, and so the resolution is t times as great as for the linkage map. This greatly increases the power to localize a gene within a candidate region at the expense of typing a larger number of markers and using methods that at this early stage are less transparent to many geneticists. With the essential properties of linearity and additivity, LD mapping is a recent invention and still in rapid evolution.

Linkage mapping depends on inference of recombination between two haplotypes in a single generation, interpreting pedigrees by a linkage map. Association mapping depends on the pattern of LD in single haplotypes, interpreted by an LDU map. Both approaches take as their prime objective the identification of a gene by determining a candidate region in the genome in the absence of information about the structure or function of the gene product. Currently single nucleotide polymorphisms (SNPs) are commonly used for association analysis because of their much greater number and ease of typing. For brevity, it is customary to pool duplications, deletions, and insertions (copy-number variations) with the much larger number of SNPs. An international effort called <a href="http://hapmap.ncbi.nlm.nih.gov/">HapMap</a> is being made to create an SNP database that will be useful for many purposes, including creation of LDU maps and choice of markers for association mapping.
 
<h2>Specific concepts/measures/formula/calculations</h2>
<b>Genetic linkage</b> is the tendency of alleles that are located close together on a chromosome to be inherited together during meiosis. Genes whose loci are nearer to each other are less likely to be separated onto different chromatids during chromosomal crossover, and are therefore said to be genetically linked. In other words, the nearer two genes are on a chromosome, the lower is the chance of a swap occurring between them, and the more likely they are to be inherited together.
<br/>

A <b>linkage map is a genetic map</b> of a species or experimental population that shows the position of its known genes or genetic markers relative to each other in terms of recombination frequency, rather than a specific physical distance along each chromosome (so a genetic map is not a physical map). The greater the frequency of recombination (segregation) between two genetic markers, the further apart they are assumed to be. Conversely, the lower the frequency of recombination between the markers, the smaller the physical distance between them. Linkage mapping is critical for identifying the location of genes that cause genetic diseases. Genetic maps have been helping researchers locate other markers, such as other genes by testing for genetic linkage of the already known markers.The place on a genetic map where a gene is located is the gene <b>locus</b> (plural loci).
<br/>

By working out the number of recombinants it is possible to obtain a measure for the distance between the genes. This distance is expressed in terms of a genetic<b> map unit (m.u.)</b>, or a <b>centimorgan</b>(after Thomas Hunt Morgan the first Drosophila geneticist) and is defined as the distance between genes for which one product of meiosis in 100 is recombinant. A <b>recombinant frequency (RF)</b> of 1% is equivalent to 1 m.u. But this equivalence is only a good approximation for small percentages; the largest percentage of recombinants cannot exceed 50%, which would be the situation where the two genes are at the extreme opposite ends of the same chromosomes. In other words, recombination rates of 50% can be achieved by genes that reside on different chromosomes simply by independent assortment, while genes that are far apart but on the same chromosome can produce similar results (if recombination occurs). In the latter situation, any crossover events would result in an exchange of genes, but only an odd number of crossover events (a 50-50 chance between even and odd number of crossover events) would result in a recombinant product of meiotic crossover (only the result of an odd number of crossovers are visible). A recombinant frequency significantly less than 50% shows that the genes are <b>linked</b>. A recombinant frequency of 50% generally means that the genes are unlinked on separate chromosomes.
<blockquote>Fun fact: Crossing over does not occur in <i>Drosophila</i> (i.e., fruit fly) males, so even genes on opposite ends of a big chromosome are completely linked in the production of male gametes.</blockquote>

Crossing-over involving three or more genes is more complicated. By adding a third gene, we now have several different types of crossing over products that can be obtained (8 possible recombinant products to be specific, as shown below). 
<img src="https://cloud.githubusercontent.com/assets/5496192/7965345/e3055780-09ea-11e5-844a-8db5e4999b46.gif" /><br/>
We first make a cross between individuals that are AABBCC and aabbcc. Next the F1 is testcrossed to an individual that is aabbcc. Using the data, the basic problem is to determine the order of the genes (there are 3 possible orders, depending on which gene is in the middle), make a map showing the distances between all pairs of genes, and determine the value of interference.
<blockquote>The concept of <b>interference</b> refers to the tendency of a "crossover" between two pairs of genes to reduce the chance that one of those genes will cross over with a different gene. <a href=
"http://classroom.synonym.com/calculate-interference-2760.html">The calculation for interference</a> involves subtracting a ratio of the observed crossover frequency to the expected crossover frequency (called the coefficient of coincidence) from one. </blockquote>
We can solve this problem following several rules:
<ol>
<li>Organize the data (test cross data showing the number of offspring of each of the 8 possible types) into reciprocal pairs. For example, a + c and + b + are a reciprocal pair, and so are a b c and + + +. The number of offspring for each member of a pair will be similar.</li>
<li>Determine which pair is the parental class: it is the LARGEST class.</li>
<li>Determine which pair is the double crossover (2CO) class: the SMALLEST class.</li>
<li>Determine which gene is in the middle. If you compare the parentals with the 2COs, the gene which switched partners is in the middle. If you think two genes have switched sides, it is the other gene that is in the middle.</li>
<li>Determine map distances for all three pairs of genes. Count the number of offspring that have had a crossover between the genes of interest, then divide by the total offspring and multiply by 100.</li>
<li>Figure the expected double crossovers: (map distance for interval I / 100 ) * (map distance for interval II / 100 ) * (total offspring) = exp 2CO.</li>
<li>Figure the interference as: I = 1 - (obs 2CO / exp 2CO). The observed 2CO class comes from the data.</li>
</ol>
LOD
LDU

<h2>References</h2>
<ul>
<li><a href="http://www.discoveryandinnovation.com/BIOL202/index.html">BIOL 202 Genetics taught by Dr. Paul SzauterUniversity of New Mexico</a>. Lecture note 7. </li>
<li>Morton, Newton E. "History of genetic mapping." Encyclopedia of Genetics, Genomics, Proteomics and Bioinformatics (2004).</li>
<li>Youtube video: <a href="https://www.youtube.com/watch?v=TU44tR0hJ8A">Genetic Recombination and Gene Mapping</a>. It explains how the frequency of recombination between linked genes can be used to determine the relative location of genes on a chromosome. Thomas Hunt Morgan and Alfred Strutevant used the fruit fly to develop a theory of chromosomal inheritance and discover crossing over.</li>
<li><a href="http://www.genomenewsnetwork.org/resources/whats_a_genome/Chp3_1.shtml">GENOME MAP</a>. Genome News Network</li>
<li><a href="http://www.ndsu.edu/pubweb/~mcclean/plsc431/linkage/linkage3.htm">Deriving Linkage Distance and Gene Order From Three-Point Crosses</a></li>
<li><a href="http://www.bios.niu.edu/johns/genetics/3ptrules.htm">Rules for solving three point crosses</a></li>
</ul>
