---
layout: post
title: "Linkage Disequlibrium (LD) mapping and coalescent"
date: 2015-04-19
---
In a finite homogeneous population all individuals will
be related to some extent. Individuals sharing a founder
mutation will have higher degree of relatedness, or IBD, in
the region around the founder mutation. In case-control
samples, increased relatedness in a region only among the
affected individuals is evidence in favor of the location of
a causative mutation in that region. Identification of
relatedness is, therefore, of great interest in medical
genetics,

Linkage disequilibrium (LD) refers to the nonrandom association of alleles in haplotypes. Such associations underlie all forms of genetic mapping (a process of establishing the locations of genes on the chromosomes), including the fine scale mapping of human disease loci.

Two loci are said to be in linkage equilibrium (LE) if their inheritance is an independent event. If the alleles at those loci are non-randomly inherited then we say that they are at linkage disequilibrium (LD). LD is most commonly caused by physical linkage of genes. When two genes are inherited on the same chromosome, depending on their distance and the likelihood of recombination between the loci they can be at high LD. However LD can be also observed due to functional interactions where even genes from different chromosomes can jointly confer evolutionary selected phenotype or can affect the viability of potential offspring. In families LD is highest because of the lowest numbers of recombination events (fewest number of meiosis events). This is especially true between inbred lines. In populations LD exists because of selection, physical closeness of the genes that causes low recombination rates or due to recent crossing or migration. On a population level, processes that influence linkage disequilibrium include genetic linkage, epistatic natural selection, rate of recombination, mutation, genetic drift, random mating, genetic hitchhiking and gene flow.

Genetic Linkage occurs when particular genetic loci or alleles for genes are inherited jointly. Genetic loci on the same chromosome are physically connected and tend to segregate together during meiosis, and are thus genetically linked.


Genetic polymorphism is the result of mutations along the branches of genealogical trees. 

LD is created by genetic drift, selection (hitchhiking?), and admixture; it is reduced by recombination and gene conversion

All existing copies of a unique
mutation can be traced back to the most-recent common
ancestor (MRCA) of that mutation. Each haplotype that
contains the mutation must also have inherited from the
MRCA a small piece of chromosome surrounding the
mutation. Different haplotypes might carry different
pieces, depending on the history of recombination. Alleles at polymorphic marker loci within the region
covered by these pieces of chromosome will be in LD with
the mutation.

Recombination is the major process that breaks down the associations between SNPs. It is unclear whether haplotype block boundaries are due to recombination hotspots, or are simply the result of recombination events that happened to occur there. If the blocks are due to hotspots, then perhaps they will be common across populations. If the blocks are due to regular recombination events, then populations may or may not share them, depending on how long ago the recombination events occurred. When large chromosomal regions are examined, the regions with high association have less recombination and less genetic variation. 

SNPs are the preferred marker type of LD mapping because of their abundancy in the genome, however, large differences in the
density of SNPs exist over the genome with regard to the genealogical process, mutation rate, selective regimes, and the  recombination and gene conversion rates per base pair.Considering whole haplotypes consisting of several adjacent
markers rather than single markers one at a time

None of the markers are individually strongly associated with phenotype.

