---
layout: post
title: "Coalescent Theory (Part 2)"
date: 2015-04-02
---

<b>Generating (artificial) sequences</b>
The number of genes required in the Wright-Fisher model for it to behave like a real population (under aforementioned assumptions) is called the <b>effective population size</b> (Ne) of that population.

<h2>Recombination and gene conversion events</h2>
When diploid individuals reproduce, there are two parents, each of which contributes one of each of its pairs of
chromosomes. One parent’s contribution is a combination of its two <a href="http://www.phschool.com/science/biology_place/labbench/lab3/homologs.html">homologous chromosomes</a> when they undergo recombination.




Recombination is the major process that breaks down the associations between SNPs. It is unclear whether haplotype block boundaries are due to recombination hotspots, or are simply the result of recombination events that happened to occur there. If the blocks are due to hotspots, then perhaps they will be common across populations. If the blocks are due to regular recombination events, then populations may or may not share them, depending on how long ago the recombination events occurred. When large chromosomal regions are examined, the regions with high association have less recombination and less genetic variation. 

Each recombination event increases the number of lineages by one, and because lineages recombine independently, the total rate of recombination when there are k lineages is kρ/2, ρ = 4Nr. Each coalescence event decreases the number of lineages by one, and the total rate of coalescence when there are k lineages is k(k − 1)/2, as we have seen previously. Since lineages are “born” at a linear rate, and “die” at a quadratic rate, the number of lineages is guaranteed to stay finite and will even hit one, occasionally (there will then temporarily be a single ancestral chromosome again

Recombination: V, allow recombination rates to vary by defining a genetic map or <a href="http://en.wikipedia.org/wiki/Recombination_hotspot">hotspots</a> along the genome

LD plays a fundamental role in the fine
scale mapping of human disease loci,

The technique known as association mapping
differs from traditional pedigree studies in that marker-disease associations
are sought in populations of unrelated individuals. To explain the idea behind
this approach, imagine that a disease causing mutation has just occurred in a
population. The chromosome on which this mutation occurred contains specific
alleles in neighboring polymorphic loci. At first, the mutation will only be
observed in conjuction with these alleles, so the association (or LD) with these
alleles will be high. Through time these associations will dissipate because of
recombination, but the closest loci will experience the fewest recombinations
and hence retain the highest levels of LD. Thus, by looking for significant correlations
between disease state and alleles, we can hope to identify the region
in which the disease causing genetic mutation lies.

Linkage disequilibrium refers to the nonrandom
association of alleles in haplotypes. Such associations
underlie all forms of genetic mapping. 



However,
whereas linkage analysis is based upon associations in
well-characterized pedigrees, LD refers to associations
within populations of ‘unrelated’individuals. There
is nonetheless a close relationship between the two
approaches, because the ‘unrelated’individuals in
a population are unrelated only in a relative and
approximate sense. In general, chromosomes sampled
from ‘unrelated’individuals in a population will be
much more distantly related than those sampled from
members of traditional pedigrees. 

This is precisely
what makes LD mapping suitable for fine-scale
mapping: there will have been more opportunities
for recombination to take place. Whereas pedigree
studies work with recombination events that
exchange megabase chunks of chromosomes, LD
studies deal with segments measured in kilobases.  When dealing with unrelated individuals,
however, the pedigree is almost completely unknown.

Genetic polymorphism is the result of mutations along the
branches of genealogical trees. 

All existing copies of a unique
mutation can be traced back to the most-recent common
ancestor (MRCA) of that mutation. Each haplotype that
contains the mutation must also have inherited from the
MRCA a small piece of chromosome surrounding the
mutation. Different haplotypes might carry different
pieces, depending on the history of recombination. Alleles at polymorphic marker loci within the region
covered by these pieces of chromosome will be in LD with
the mutation. 
